lib/SILOptimizer/Mandatory/TransferNonSendable.cpp

```diff
@@ -242,6 +250,56 @@ struct TermArgSources {
     for (auto pair : llvm::enumerate(valueRange))
       argSources.insert(destBlock->getArgument(pair.index()), pair.value());
   }
+
+  TermArgSources() {}
+
+  void init(SILInstruction *inst) {
+    switch (cast<TermInst>(inst)->getTermKind()) {
+    case TermKind::UnreachableInst:
+    case TermKind::ReturnInst:
+    case TermKind::ThrowInst:
+    case TermKind::ThrowAddrInst:
+    case TermKind::YieldInst:
+    case TermKind::UnwindInst:
+    case TermKind::TryApplyInst:
+    case TermKind::SwitchValueInst:
+    case TermKind::SwitchEnumInst:
+    case TermKind::SwitchEnumAddrInst:
+    case TermKind::AwaitAsyncContinuationInst:
+    case TermKind::CheckedCastAddrBranchInst:
+      llvm_unreachable("Unsupported?!");
+
+    case TermKind::BranchInst:
+      return init(cast<BranchInst>(inst));
+
+    case TermKind::CondBranchInst:
+      return init(cast<CondBranchInst>(inst));
+
+    case TermKind::DynamicMethodBranchInst:
+      return init(cast<DynamicMethodBranchInst>(inst));
+
+    case TermKind::CheckedCastBranchInst:
+      return init(cast<CheckedCastBranchInst>(inst));
+    }
+
+    llvm_unreachable("Covered switch isn't covered?!");
+  }
+
+private:
+  void init(BranchInst *bi) { addValues(bi->getArgs(), bi->getDestBB()); }
```

Comment:
Is it really worth having separate overloads of init()? Why not just fold them into the switch

Reply:
I think I was just moving code around. You are right, there really isn't a purpose of it.